homeassistant/components/switch/wemo.py

```diff
@@ -109,23 +120,88 @@ def device_state_attributes(self):
 
         if self.insight_params or (self.coffeemaker_mode is not None):
             attr[ATTR_CURRENT_STATE_DETAIL] = self.detail_state
+            attr[ATTR_POWER_CURRENT_W] = self.power_current_watt
+            # attr[ATTR_POWER_AVG_W] = self.power_average_watt
+            attr[ATTR_POWER_TODAY_MW_MIN] = self.power_today_mw_min
+            attr[ATTR_POWER_TOTAL_MW_MIN] = self.power_total_mw_min
+            attr[ATTR_ON_FOR_TIME] = self.on_for
+            attr[ATTR_ON_TODAY_TIME] = self.on_today
+            attr[ATTR_ON_TOTAL_TIME] = self.on_total
+            attr[ATTR_POWER_THRESHOLD] = self.power_threshold
 
         if self.coffeemaker_mode is not None:
             attr[ATTR_COFFEMAKER_MODE] = self.coffeemaker_mode
 
         return attr
 
+#    @property
+    def _current_power_mw(self):
+        """Current power usage in mW."""
+        if self.insight_params:
+            try:
+                return self.insight_params['currentpower']
+            except KeyError:
+                return None
+
+    @property
+    def power_current_watt(self):
+        """Current power usage in W."""
+        if self.insight_params:
+            try:
+                return self._current_power_mw() / 1000
+            except TypeError:
+                return None
+
+    @property
+    def power_threshold(self):
+        """Threshold of W at which Insight will indicate it's load is ON."""
+        if self.insight_params:
+            return self.insight_params['powerthreshold'] / 1000
+
+    @staticmethod
+    def as_uptime(_seconds):
+        """Format seconds into uptime string in the format: 00d 00h 00m 00s."""
+        uptime = datetime(1, 1, 1) + timedelta(seconds=_seconds)
+        return "{:0>2d}d {:0>2d}h {:0>2d}m {:0>2d}s".format(uptime.day-1,
+                                                            uptime.hour,
+                                                            uptime.minute,
+                                                            uptime.second)
+
+    @property
+    def on_for(self):
+        """On time in seconds."""
+        if self.insight_params:
+            return WemoSwitch.as_uptime(self.insight_params['onfor'])
+
+    @property
+    def on_today(self):
+        """On time in seconds."""
+        if self.insight_params:
+            return WemoSwitch.as_uptime(self.insight_params['ontoday'])
+
+    @property
+    def on_total(self):
+        """On time in seconds."""
+        if self.insight_params:
+            return WemoSwitch.as_uptime(self.insight_params['ontotal'])
+
     @property
-    def current_power_mwh(self):
-        """Current power usage in mWh."""
+    def power_total_mw_min(self):
```

Comment:
The Insight returns average milliwatt power consumed for each minute.  So for 12W it will increment this value by 12,000 each minute.  So it's mW/min.

mWh is average milliwatts per hour...  The Insight doesn't provide that.